app/controllers/api/v2x0/taggings_controller.rb

```diff
@@ -0,0 +1,38 @@
+module Api
+  module V2x0
+    class TaggingsController < ApplicationController
+      include Api::V1::Mixins::IndexMixin
+
+      # Present these as tags
+      private_class_method def self.api_doc_definition
+        @api_doc_definition ||= api_doc.definitions["Tag"]
+      end
+
+      def self.presentation_name
+        "Tag".freeze
+      end
+
+      def create
+        primary_instance = primary_collection_model.find(request_path_parts["primary_collection_id"])
+        tag = Tag.find_or_create_by!(Tag.parse(params_for_create["tag"]))
+        primary_instance.tags << tag
+
+        render :json => tag, :status => :created, :location => "#{instance_link(primary_instance)}/tags"
+      end
+
+      def destroy
+        primary_instance = primary_collection_model.find(request_path_parts["primary_collection_id"])
+        tag = Tag.find_by!(Tag.parse(body_params["tag"]))
```

Comment:
Found when adding functionality to Catalog (https://github.com/RedHatInsights/catalog-api/pull/495):
OpenAPI is not ok with having `DELETE` operations that use request bodies, it's not part of the HTTP spec. It also can't be tested with rspec, per here: https://github.com/zipmark/rspec_api_documentation/issues/368

Not sure where we want to go with this one, maybe just allow deleting on the subresource?

Reply:
well, HTTP spec does not prohibit it.  There seems to be a PR for the OpenAPI 3 https://github.com/OAI/OpenAPI-Specification/pull/1937 to add support for it.

Reply:
> It also can't be tested with rspec

I was able to add tests [here](https://github.com/RedHatInsights/topological_inventory-api/pull/259/commits/e1f3321dbbd3282e43c35335d9e4e9f363bf125e#diff-af8561cfcf1754b19f858109d77eca64R45)

Reply:
> Not sure where we want to go with this one, maybe just allow deleting on the subresource?

I agree that it is unconventional to DELETE with a request body.  Unfortunately the user doesn't know the id of the tag join record that we're displaying.  (and I don't think they should see it either)

Reply:
@bdunne looking at your specs showed me where I was wrong - when you pass `:params` a hash in rspec on a POST it implicitly must call `.to_json` on it, when I specify that explicitly for my DELETE requests my specs work again. That makes me feel so much better, I'm ok with it. 

Hopefully that change gets merged so we can have the requestBody defined in the OpenAPI schema for the DELETE operation.